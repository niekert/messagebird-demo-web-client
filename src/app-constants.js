// Status codes
export const INITIAL = 'INITIAL';
export const PENDING = 'PENDING';
export const OK = 'OK';
export const ERROR = 'ERROR';
