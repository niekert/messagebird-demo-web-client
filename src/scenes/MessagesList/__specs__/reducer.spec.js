import { INITIAL } from 'app-constants';
import {
  FETCH_MESSAGES_LIST,
  FETCH_MESSAGES_LIST_SUCCESS,
  FETCH_MESSAGES_LIST_ERROR,
  ADD_NEW_MESSAGE,
} from '../actions';
import reduce from '../reducer';

const initialState = {
  status: INITIAL,
  messages: [],
  nextUrl: null,
  error: null,
};

describe('messagesListReducer', () => {
  test(`${FETCH_MESSAGES_LIST}`, () => {
    const result = reduce(initialState, {
      type: FETCH_MESSAGES_LIST,
    });

    expect(result).toMatchSnapshot();
  });

  test(`${FETCH_MESSAGES_LIST_SUCCESS}`, () => {
    const result = reduce(initialState, {
      type: FETCH_MESSAGES_LIST_SUCCESS,
      payload: {
        messages: 'mock_messages',
        nextUrl: 'next_url',
      },
    });

    expect(result).toMatchSnapshot();
  });

  test(`${FETCH_MESSAGES_LIST_ERROR}`, () => {
    reduce(initialState, {
      type: FETCH_MESSAGES_LIST_SUCCESS,
      payload: {
        error: new Error('fetch_messages_error'),
      },
    });
  });

  test(`${ADD_NEW_MESSAGE}`, () => {
    const currentState = { ...initialState, messages: ['test'] };
    reduce(currentState, {
      type: ADD_NEW_MESSAGE,
      payload: {
        message: 'wow second message wow',
      },
    });
  });
});
