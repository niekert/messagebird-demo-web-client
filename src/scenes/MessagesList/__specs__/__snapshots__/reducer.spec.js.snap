// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messagesListReducer FETCH_MESSAGES_LIST 1`] = `
Object {
  "error": null,
  "messages": Array [],
  "nextUrl": null,
  "status": "PENDING",
}
`;

exports[`messagesListReducer FETCH_MESSAGES_LIST_SUCCESS 1`] = `
Object {
  "error": null,
  "messages": "mock_messages",
  "nextUrl": "next_url",
  "status": "OK",
}
`;
